vendor/bat-native-ads/src/bat/ads/internal/security/crypto_util.cc

```diff
@@ -162,42 +120,83 @@ std::vector<uint8_t> Sha256Hash(const std::string& value) {
   return sha256;
 }
 
-bool Verify(const ConfirmationInfo& confirmation) {
-  std::string credential;
-  base::Base64Decode(confirmation.credential, &credential);
-
-  base::Optional<base::Value> value = base::JSONReader::Read(credential);
-  if (!value || !value->is_dict()) {
-    return false;
+KeyPairInfo GenerateSignKeyPairFromSeed(const std::vector<uint8_t>& seed) {
+  KeyPairInfo key_pair;
+  if (seed.empty()) {
+    return key_pair;
   }
 
-  base::DictionaryValue* dictionary = nullptr;
-  if (!value->GetAsDictionary(&dictionary)) {
-    return false;
-  }
+  std::vector<uint8_t> secret_key = seed;
+  secret_key.resize(crypto_sign_SECRETKEYBYTES);
+  std::vector<uint8_t> public_key(crypto_sign_PUBLICKEYBYTES);
+  crypto_sign_keypair(&public_key.front(), &secret_key.front(), 1);
 
-  const std::string* signature = dictionary->FindStringKey("signature");
-  if (!signature) {
-    return false;
-  }
+  key_pair.public_key = public_key;
+  key_pair.secret_key = secret_key;
 
-  VerificationSignature verification_signature =
-      VerificationSignature::decode_base64(*signature);
-  if (privacy::ExceptionOccurred()) {
-    NOTREACHED();
-    return false;
-  }
+  return key_pair;
+}
+
+KeyPairInfo GenerateBoxKeyPair() {
+  std::vector<uint8_t> public_key(crypto_box_PUBLICKEYBYTES);
+  std::vector<uint8_t> secret_key(crypto_box_SECRETKEYBYTES);
+  crypto_box_keypair(&public_key.front(), &secret_key.front());
+
+  KeyPairInfo key_pair;
+  key_pair.public_key = public_key;
+  key_pair.secret_key = secret_key;
 
-  const std::string payload = CreateConfirmationRequestDTO(confirmation);
+  return key_pair;
+}
 
-  UnblindedToken unblinded_token = confirmation.unblinded_token.value;
-  VerificationKey verification_key = unblinded_token.derive_verification_key();
-  if (privacy::ExceptionOccurred()) {
-    NOTREACHED();
-    return false;
+std::vector<uint8_t> GenerateSecretKeyFromSeed(const std::string& seed_base64) {
+  std::string seed;
+  if (!base::Base64Decode(seed_base64, &seed)) {
+    return {};
   }
 
-  return verification_key.verify(verification_signature, payload);
+  const std::vector<uint8_t> derived_key = GetHKDF(seed);
+  KeyPairInfo key_pair = GenerateSignKeyPairFromSeed(derived_key);
+
+  return key_pair.secret_key;
+}
+
+std::vector<uint8_t> GenerateNonce() {
+  std::vector<uint8_t> nonce(crypto_box_NONCEBYTES);
+  base::RandBytes(&nonce.front(), nonce.size());
+
+  return nonce;
+}
+
+std::vector<uint8_t> Encrypt(const std::vector<uint8_t>& plaintext,
+                             const std::vector<uint8_t>& nonce,
+                             const std::vector<uint8_t>& public_key,
+                             const std::vector<uint8_t>& ephemeral_secret_key) {
+  // API requires 32 leading zero-padding bytes
```

Comment:
nit: do we need?

Reply:
Hm good q: I've seen the comment across github on code examples. It is in the c lib spec but we have different input and output paddings since JS spec is different. All in all it cost me a day to figure this out so thought I explicitly highlight it. Wdyt?

Reply:
Sounds good, can we just add the name of the API before API or do you think it is clear that nacl is the API?

Reply:
+1 to keeping it in the code. The C lib is very opaque without a background in its functionality. More explainers the better in this case IMO.